client/src/views/workspaces/WorkspacesPage.vue

```diff
@@ -483,6 +551,11 @@ function onMsSorterChange(event: MsSorterChangeEvent): void {
   sortByAsc.value = event.sortByAsc;
 }
 
+async function onWorkspaceMenuClick(selectedMenu: WorkspaceMenu): Promise<void> {
+  setWorkspaceMenu(selectedMenu);
+  await navigateTo(Routes.Workspaces, { params: { handle: getConnectionHandle() } });
+}
+
```

Comment:
You don't need to navigate anymore, this function can be deleted and you can just call `setWorkspaceMenu` instead